File: Progamming Languages/C#/OOP/exercise_23/Line.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace exercise_23
{
    public class Line:Point
    {
        public Point Begin{
            get; set;
        }

        public Point End{
            get; set;
        }
        public Line(Point begin, Point end){
            Begin = begin;
            End = end;
        }

        public Line( int x1, int y1, int x2, int y2){
            Begin.X = x1;
            Begin.Y = y1;
            End.X = x2;
            End.Y = y2;
        }
        public double getLength(){
            return Math.Sqrt(Math.Pow((End.X - Begin.X), 2) + Math.Pow((End.Y - Begin.Y) , 2));
        }
    }
}

File: Progamming Languages/Java/Test/ex2/Line.java
import java.io.*;
import java.math.*;
public class Line extends Shape {
    public Point p1, p2;
    private BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

    public Line() throws Exception{
        System.out.println("Enter Two Points Of The Line:");
        this.p1 = new Point();
        this.p2 = new Point();
    }
    public void ShapeName(){
        System.out.println(">>> LINE");
    }
    public double LengthOfLine(){
        return Math.sqrt(Math.pow(Math.abs(p2.x - p1.x), 2) + Math.pow(Math.abs(p2.y - p1.y), 2));
    }
    public void Display(){
        this.p1.Display();
        this.p2.Display();
    }
    public double Area(){
        return 0.0;
    }
}
